presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -106,9 +110,9 @@
     private final AtomicReference<Supplier<OperatorInfo>> infoSupplier = new AtomicReference<>();
     private final boolean collectTimings;
 
-    // memoryRevokingRequestedFuture is done iff memory revoking was requested for operator
     @GuardedBy("this")
-    private SettableFuture<?> memoryRevokingRequestedFuture = SettableFuture.create();
+    private final AtomicBoolean memoryRevokingRequested = new AtomicBoolean();
+    private final AtomicReference<Runnable> memoryRevokingRequestedListener = new AtomicReference<>();
```

Comment:
Same here: make it `@Nullable @GuardedBy("this") Runnable memoryRevokingRequestedListener` or `@GuardedBy("this") Optional<Runnable> memoryRevokingRequestedListener`